pandas/core/indexers/objects.py

```diff
@@ -102,7 +102,7 @@ def get_window_bounds(
         closed: str | None = None,
         step: int | None = None,
     ) -> tuple[np.ndarray, np.ndarray]:
-        if center:
+        if center or self.window_size == 0:
```

Comment:
Wouldn't this result in the same resulting `offset` as before? i.e. 

```python
In [1]: 0 - 1 // 2
Out[1]: 0
```

Reply:
In your code `1 // 2` is getting evaluated first. The parentheses matter

```python
>>> (0 - 1) // 2
-1
```

Reply:
I verified this fixed the segfault via a local run of #55102 - hopefully can make progress soon to getting that up in CI since this is now 2 releases in a row where we have a segfault when cutting the release.

#55151 is a big step towards getting UBSAN/ASAN set up as the datetime code issues a ton of errors, and by design those tools stop at the first error encountered. If you have any time to help review that and push through would appreciate it